spec/solver_spec.rb

```diff
@@ -0,0 +1,67 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Factorial Function Unit Testing'
+  it 'Should return 120' do
+    solver = Solver.new
+    result = solver.factorial(5)
+    expect(result).to eq(120)
+  end
+
+  it 'Should return 6' do
+    solver = Solver.new
+    result = solver.factorial(3)
+    expect(result).to eq(6)
+  end
+  it 'Should return 40,320' do
+    solver = Solver.new
+    result = solver.factorial(8)
+    expect(result).to eq(40_320)
+  end
+
+  it 'Negative number factorial should rasie an error' do
+    expect { solver.factorial(-1) }.to raise_error(Exception)
+  end
+end
```

Comment:
- [ ] Kindly add a test for the special case scenario when the number is 0 - its result should be 1.